Balloonduino.ino

```diff
@@ -1,1196 +1,40 @@
-     */
-    if (!rtc.begin())
-    {
-        debug_serial.println("WARNING: DS1308 initialization failure.");
-    }
-    else
-    {
-        debug_serial.println("DS1308 initialized.");
-    }
-
-    if (!rtc.isrunning())
-    {
-        debug_serial.println("WARNING: RTC is NOT running!");
-        // following line sets the RTC to the date & time this sketch was compiled
-        rtc.adjust(DateTime(F(__DATE__), F(__TIME__)));
-        // This line sets the RTC with an explicit date & time, for example to set
-        // January 21, 2014 at 3am you would call:
-        // rtc.adjust(DateTime(2014, 1, 21, 3, 0, 0));
-    }
-
-    //// SoftRTC (for subsecond precision)
-    SoftRTC.begin(rtc.now());    // Initialize SoftRTC to the current time
-    start_millis = millis();    // get the current millisecond count
-
-    //// Init BME
-    if (!bme.begin(0x76))
-    {
-        debug_serial.println("WARNING: BME280 initialization failure.");
-    }
-    else
-    {
-        debug_serial.println("BME280 initialized.");
-    }
-
-    //// Init ADS
-    ads.begin();
-    ads.setGain(GAIN_ONE);
-    debug_serial.println("ADS1015 initialized.");
-
-    //// Init SD card
-    SPI.begin();
-    pinMode(53, OUTPUT);
-    if (!SD.begin(53))
-    {
-        debug_serial.println("SD Card initialization failure.");
-    }
-    else
-    {
-        debug_serial.println("SD Card initialized.");
-    }
-
-    // xbee
-    debug_serial.println("Beginning xbee init");
-
-    int xbeeStatus = InitXBee(XBEE_ADDR, PAN_ID, xbee_serial, false);
-    if (!xbeeStatus)
-    {
-        debug_serial.println("XBee initialized.");
-    }
-    else
-    {
-        debug_serial.print("ERROR: XBee initialization error with code: ");
-        debug_serial.println(xbeeStatus);
-    }
-
-    //// Init SSC
-    //  set min / max reading and pressure, see datasheet for the values for your
-    //  sensor
-    ssc.setMinRaw(0);
-    ssc.setMaxRaw(16383);
-    ssc.setMinPressure(0.0);
-    ssc.setMaxPressure(30);
-
-    //  start the sensor
-    debug_serial.print("SSC start: ");
-    debug_serial.println(ssc.start());
-
-    // MicroSD
-    // appends to current file
-    // NOTE: Filenames must be shorter than 8 characters
-    File IMULogFile = SD.open("IMU_LOG.txt", FILE_WRITE);
-    File PWRLogFile = SD.open("PWR_LOG.txt", FILE_WRITE);
-    File ENVLogFile = SD.open("ENV_LOG.txt", FILE_WRITE);
+    balloonduino.begin();
 }
 
-void loop(void)
+void loop()
 {
-    /*  loop()
-     *
-     *  Reads sensor
-     *  Log sensors
-     *  Reads from xbee and processes any data
-     */
-
-// declare structures to store data
-    IMUData_s IMUData;
-    PWRData_s PWRData;
-    ENVData_s ENVData;
-
-// increment read counters
+    // increment read counters
     imu_read_ctr++;
     pwr_read_ctr++;
     env_read_ctr++;
 
-// read sensors if time between last read
+    // read sensors if time between last read
     if (imu_read_ctr > imu_read_lim)
```

Comment:
rather than having the user keep track of counters, I think it would be easier to have a function along the lines of setIMUPktInterval(uint16 milli_sec_period) which would set a property of the class and then have a function "bool isTimeToLogIMU()" which returns true if the time since the last log entry (as recorded by log_imu) has ellapsed and indicates to the user that they need to log the data